src/ui/public/index_patterns/index_patterns.js

```diff
@@ -41,6 +41,7 @@ export function IndexPatternsProvider(Notifier, Private) {
   self.cache = patternCache;
   self.getIds = getProvider('id');
   self.getTitles = getProvider('attributes.title');
+  self.getIndexPatterns = getProvider();
```

Comment:
I probably don't know enough of the history, but why wasn't the original implementation consistent with response type? Why can't saved objects be converted into `IndexPattern` objects? Is it an issue of performance? I don't really understand the id collision comment I mentioned earlier either